base/linalg/uniformscaling.jl

```diff
@@ -167,6 +167,8 @@ end
 
 inv(J::UniformScaling) = UniformScaling(inv(J.λ))
 norm(J::UniformScaling, p::Real=2) = abs(J.λ)
+det(J::UniformScaling{T}) where {T} = isone(J.λ) ? one(T) : throw(ArgumentError("Determinant of UniformScaling is only well-defined when λ=1."))
```

Comment:
`det(eye(Int, 3, 3))` is a `Float64` only because it is promoted to perform an LU factorization. In this case we don't need that so I think `one(T)` is perfectly fine. We have for example:
```julia
julia> det(Diagonal([1,2,3]))
6
```
for the same reason; no need to promote for LU.